frontend/ui/impl/src/test/java/com/virtuslab/gitmachete/frontend/ui/impl/GitMacheteErrorReportSubmitterTest.java

```diff
@@ -0,0 +1,49 @@
+package com.virtuslab.gitmachete.frontend.ui.impl;
+
+import static org.mockito.ArgumentMatchers.*;
+
+import java.util.HashMap;
+
+import com.intellij.diagnostic.AbstractMessage;
+import com.intellij.diagnostic.IdeaReportingEvent;
+import com.intellij.diagnostic.LogMessage;
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({GitMacheteErrorReportSubmitter.class})
+public class GitMacheteErrorReportSubmitterTest {
+  @Test
+  public void shouldNotThrowWhenGettingBugTemplate() throws Exception {
+    GitMacheteErrorReportSubmitter reportSubmitter = new GitMacheteErrorReportSubmitter();
+
+    Whitebox.invokeMethod(reportSubmitter, "getBugTemplate");
+  }
+
+  @Test
+  public void testGitHubIssueUriConstruction() throws Exception {
+    IdeaLoggingEvent[] events = new IdeaLoggingEvent[1];
+
+    AbstractMessage messageObject = Whitebox.newInstance(LogMessage.class);
+
+    events[0] = new IdeaReportingEvent(messageObject, /* message */ null, /* stacktrace */"", /* plugin */null);
+
+    GitMacheteErrorReportSubmitter reportSubmitter = PowerMockito.spy(new GitMacheteErrorReportSubmitter());
```

Comment:
Is `spy` needed here? AFAICS all other tests use `mock(...)`

Reply:
With spy you can to partial mocking [Mockito API](https://javadoc.io/static/org.mockito/mockito-core/4.6.1/org/mockito/Mockito.html#spy), mock does not let you call real methods.

Reply:
Oh, so that's basically needed to call private methods on `GitMacheteErrorReportSubmitter`? 🤔 As in, assuming all tested methods of `GitMacheteErrorReportSubmitter` were public, neither `spy()` nor `mock()` would be needed?

Reply:
Wow okay... pls add a comment about the usage of `spy` into the code then!

Reply:
The link in my first answer was incorrect, sorry -> [Mockito API](https://javadoc.io/static/org.mockito/mockito-core/4.6.1/org/mockito/Mockito.html#spy)